### PR TITLE
Docs: Mention how to add new functionality without breaking APIs

### PR DESCRIPTION
**Pull request by nastra**

### CODE REVIEW

**Review by rdblue — APPROVED** [review]
Seems fine to me.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -108,6 +108,85 @@ Example:
...
+
+Let's assume we'd want to add a `createBranch(String name)` method to the `ManageSnapshots` API.
+
+The most straight-forward way would be to add the below code:
+
+```java
+public interface ManageSnapshots extends PendingUpdate<Snapshot> {
+  // existing code...
+
+  // introduces an API-breaking change
```
Comment (mystic-lama) [inline CONTRIBUTING.md:122]: Minor nit: can we say

// introduce an API-breaking change by adding the method
or
// introduce an API-breaking change by adding the below method

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -108,6 +108,85 @@ Example:
...
+  new: method org.apache.iceberg.ManageSnapshots org.apache.iceberg.ManageSnapshots::createBranch(java.lang.String)
+
+  SOURCE: BREAKING, BINARY: NON_BREAKING, SEMANTIC: POTENTIALLY_BREAKING
+
+  From old archive: <none>
+  From new archive: iceberg-api-1.4.0-SNAPSHOT.jar
+
+  If this is an acceptable break that will not harm your users, you can ignore it in future runs like so for:
+
+    * Just this break:
```
Comment (mystic-lama) [inline CONTRIBUTING.md:165]: newbie question: If we introduce a new API, the CI shall flag it. How does it get handled during automated build?

**Review by mystic-lama — COMMENTED** [review]
+1
This is very helpful

File: CONTRIBUTING.md (hunk 2 above)
Comment (nastra) [reply CONTRIBUTING.md:165]: it is handled the same way as on CI. `revapi` is being executed as part of the normal build, which then flags API/ABI-breaking changes

**Review by ConeyLiu — APPROVED** [review]
+1, thanks @nastra for the guideline.

File: CONTRIBUTING.md (hunk 1 above)
Comment (nastra) [reply CONTRIBUTING.md]: done

**Comment by nastra** [comment]
thanks for the reviews everyone

**Pull request merged**